ggshield/core/errors.py

```diff
@@ -60,6 +60,15 @@ def __init__(self, message: str):
         super().__init__(ExitCode.UNEXPECTED_ERROR, message)
 
 
+class InvalidGitRefError(_ExitError):
+    """
+    Raised when the git reference does not exist
+    """
+
+    def __init__(self, ref: str):
+        super().__init__(ExitCode.USAGE_ERROR, f"Not a git reference: {ref}.")
```

Comment:
Is it correct for an Error class to expect a string argument that is not the error message? This is already the case for some other classes, for instance `UnknownInstanceError`.

WDYT @agateau-gg?

Reply:
I think it's OK because at the point the exception is raised we don't have any other information we could provide.

I don't understand what we gain from adding this exception class though: do you plan to catch it specifically in calling code, or is it going to be raised from other places? If not then the code could stay as it was.

Reply:
We used to call `UsageError` raising an exception which was not of type `_ExitError`. However, we are using the function `handle_exception` from this same file `ggshield/core/errors.py`, which tests if the exception raised is from type `_ExitError` else return an `UNEXPECTED_ERROR`. So with the old way and after adding the exception wrapper, we always ended up having this exit code whereas we wanted a `USAGE_ERROR` one.
Do you think there is a better way to handle this situation?

Reply:
Oh right, good catch! No, I can't think of a better way to handle this :+1: (your explanation would make a great commit message, though :wink:).

Reply:
Noted, I'll think about it next time. Thanks for your review :)